pkg/master/pod_cache.go

```diff
@@ -221,6 +227,10 @@ func (p *PodCache) GarbageCollectPodStatus() {
 // calling again, or risk having new info getting clobbered by delayed
 // old info.
 func (p *PodCache) UpdateAllContainers() {
+	// TODO: this is silly, we should pro-actively update the pod status when
+	// the API server makes changes.
+	p.clearNodeStatus()
```

Comment:
The node status isn't updated because we just log the error and move on:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/master/pod_cache.go#L242

We could certainly set Unknown there too, but we'll see pods flicker in and out in the case of transient failures.  (we would likely want to add some sort of error count to the cache.)

I updated the replication controller to filter Unknown pods also.

I will push a new cluster and see if this works correctly 



We could set unknown there too if we wanted to.